akka-docs/src/main/paradox/stream/operators/Source-or-Flow/mapAsyncPartitioned.md

```diff
@@ -0,0 +1,130 @@
+# mapAsyncPartitioned
+
+Pass incoming elements to a function that extracts a partitioning key from the element, then to a function that returns a @scala[`Future`] @java[`CompletionStage`] result, bounding the number of incomplete @scala[futures] @java[`CompletionStage` s] per partitioning key.
+
+@ref[Asynchronous operators](../index.md#asynchronous-operators)
+
+## Signature
+
+@apidoc[Source.mapAsyncPartitioned](Source) { scala="#mapAsyncPartitioned[T,P](parallelism:Int,perPartition:Int)(partitioner:Out=%3EP)(f:(Out,P)=%3Escala.concurrent.Future[T]):FlowOps.this.Repr[T]" java="#mapAsyncPartitioned(int,int,akka.japi.function.Function,java.util.function.BiFunction" }
+@apidoc[Flow.mapAsyncPartitioned](Flow) { scala="#mapAsyncPartitioned[T,P](parallelism:Int,perPartition:Int)(partitioner:Out=%3EP)(f:(Out,P)=%3Escala.concurrent.Future[T]):FlowOps.this.Repr[T]" java="#mapAsyncPartitioned(int,int,akka.japi.function.Function,java.util.function.BiFunction" }
+
+## Description
+
+Pass incoming elements to a function that assigns a partitioning key, then to function that returns a @scala[`Future`] @java[`CompletionStage`] result.  When the @scala[future] @java[`CompletionStage`] completes successfully, the result is passed downstream.
+
+Up to `parallelism` futures can be processed concurrently; additionally no more than `perPartition` @scala[futures] @java[`CompletionStage` s] with the same partitioning key will be incomplete at any time.
+
+Regardless of completion order, results will be emitted in order of the incoming elements which gave rise to the @scala[future] @java[`CompletionStage`].
```

Comment:
some inconsistent "future" and `Future`

Reply:
below it is mentioned equivalent to mapAsyncUnordered, which then doesn't align with this statement of "order of the incoming elements"

Reply:
ah, you mean `mapAsyncUnordered(parallelism = 1)`. That is just confusing to mention, because that is the same as `mapAsync(1)` and is ordered.